internal/sim/executor.go

```diff
@@ -384,10 +384,17 @@ func (e *executor) call(caller string, replica int, reg *codegen.Registration, m
 		strings[i] = fmt.Sprint(arg)
 	}
 
+	// Extract the trace id.
+	traceID, _ := extractIDs(ctx)
+	if traceID == 0 {
+		// TODO(mwhittaker): Link to online documentation with better
+		// explanation of this error.
+		panic(fmt.Errorf("missing simulation trace context. Make sure you are correctly propagating contexts through your workload."))
```

Comment:
Can you be more explicit right now. In particular, there's something we require the user to do (e.g., propagate the contexts through to the component calls), might as well mention it briefly here?

Reply:
Reworded to the error to hopefully be clearer.